### PR TITLE
docker: add workaround for CloudFront

### PR DESCRIPTION
**Pull request by giuseppe**
CloudFront returns an unquoted ':' in the boundary, special handle it.

It is needed for partial pulls from quay.io.
    
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: docker/docker_image_src.go [hunk 1]
```diff
@@ -399,7 +405,22 @@ func (s *dockerImageSource) GetBlobAt(ctx context.Context, info types.BlobInfo,
 	case http.StatusPartialContent:
 		mediaType, params, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
 		if err != nil {
-			return nil, nil, err
+			if err == mime.ErrInvalidMediaParameter {
+				// CloudFront returns an invalid MIME type, that contains an unquoted ":", let's
+				// handle it here.
+				re := regexp.MustCompile("multipart/byteranges; boundary=([A-Za-z-0-9:]+)")
```
Comment (mtrmac) [inline docker/docker_image_src.go:411]: This can be compiled just once per process lifetime, can’t it?

File: docker/docker_image_src.go [hunk 2]
```diff
@@ -399,7 +405,22 @@ func (s *dockerImageSource) GetBlobAt(ctx context.Context, info types.BlobInfo,
 	case http.StatusPartialContent:
 		mediaType, params, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
 		if err != nil {
-			return nil, nil, err
+			if err == mime.ErrInvalidMediaParameter {
+				// CloudFront returns an invalid MIME type, that contains an unquoted ":", let's
```
Comment (mtrmac) [inline docker/docker_image_src.go:409]: I don’t quite see how this comment `:` relates to the code, but, I guess we’ll see from the test case…

Do you mean that it is invalid to have a `;` in the _boundary_? There is no `:` in `multipart/byteranges`.

**Review by mtrmac — COMMENTED** [review]
Thanks.

Most importantly, please add a unit test, ideally with an unmodified response of the potential server. (If that is easier when splitting the content-type parser into a separate function, sure, why not.)

We’ve already been in a situation where we have a workaround and no way to maintain it because we don’t know what the workaround is looking for how what are the essential parts, let’s not add any such situations.

File: docker/docker_image_src.go (hunk 2 above)
Comment (mtrmac) [reply docker/docker_image_src.go:409]: Ah, the PR description does say “boundary”.

**Comment by giuseppe** [comment]
@mtrmac thanks for the review.  I addressed your comments in the last version :arrow_up:

**Comment by rhatdan** [comment]
LGTM

**Review by mtrmac — COMMENTED** [review]
LGTM. Thanks again!

**Pull request merged**